crates/store/re_chunk/src/helpers.rs

```diff
@@ -0,0 +1,360 @@
+use std::sync::Arc;
+
+use arrow2::array::Array as ArrowArray;
+
+use re_log_types::{TimeInt, Timeline};
+use re_types_core::{Component, ComponentName, SizeBytes};
+
+use crate::{Chunk, ChunkResult, RowId};
+
+// --- Helpers ---
+
+impl Chunk {
+    // --- Batch ---
+
+    /// Returns the raw data for the specified component.
+    ///
+    /// Returns an error if the row index is out of bounds.
+    #[inline]
+    pub fn component_batch_raw(
+        &self,
+        component_name: &ComponentName,
+        row_index: usize,
+    ) -> Option<ChunkResult<Box<dyn ArrowArray>>> {
```

Comment:
Here, and in all the signatures below, I think we should try to keep result types on the outside as in: `ChunkResult<Option<Box<dyn ArrowArray>>>` 

This seems like it lets us handle serialization errors consistently and then end up with `Option<Box<dyn ArrowArray>>` inline with:
- https://github.com/rerun-io/rerun/issues/6602

Reply:
I had `ChunkResult<Option<Box<dyn ArrowArray>>>` at the beginning, and it was just as awkward but for different reasons.
In the end I ended up with that variant, but ultimately I don't really have a strong opinion for one or the other.

Two notes, though:
* I purposefully ignored / stayed away from #6602, as I quickly realized that this PR stream would be complicated enough without attempting to modify our existing semantics. I don't mind revisiting that in the future, but for now I don't think this should have any bearing on the current PR.
* I wanted these Chunk APIs to be low-level enough so that more high-level semantics could be built on top of them for the  common cases. These high-level APIs are introduced later in this PR stream, at the `re_query` layer.